File: example/src/main/java/io/github/stavshamir/swagger4kafka/example/consumers/ExampleConsumer.java
package io.github.stavshamir.swagger4kafka.example.consumers;

import io.github.stavshamir.swagger4kafka.example.dtos.ExamplePayloadDto;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.kafka.annotation.KafkaListener;
import org.springframework.stereotype.Service;

@Service
public class ExampleConsumer {

    private static Logger logger = LoggerFactory.getLogger(ExampleConsumer.class);

    @KafkaListener(topics = "example-topic", containerFactory = "exampleKafkaListenerContainerFactory")
    public void receiveExamplePayload(ExamplePayloadDto payload) {
        logger.info("Received new message in example-topic: {}", payload.toString());
    }

}

File: swagger4kafka-integration-tests/src/main/kotlin/io/github/stavshamir/swagger4kafka/integration/consumers/ExampleConsumer.kt
package io.github.stavshamir.swagger4kafka.integration.consumers

import io.github.stavshamir.swagger4kafka.integration.dtos.ExamplePayloadDto
import org.slf4j.LoggerFactory
import org.springframework.kafka.annotation.KafkaListener
import org.springframework.stereotype.Service

@Service
class ExampleConsumer {

    companion object {
        private val logger = LoggerFactory.getLogger(ExampleConsumer::class.java)
    }

    @KafkaListener(topics = ["example-topic"], containerFactory = "exampleKafkaListenerContainerFactory")
    fun receiveExamplePayload(payload: ExamplePayloadDto) {
        logger.info("Received new message in example-topic: {}", payload.toString())
    }

}